ruleset.go

```diff
@@ -132,37 +133,24 @@ type RuleActionExtraction struct {
 	Regex  string `json:"regex,omitempty"`
 }
 
-// ListRulesets gets all rulesets.
-func (c *Client) ListRulesets() (*ListRulesetsResponse, error) {
-	rulesetResponse := new(ListRulesetsResponse)
-	rulesets := make([]*Ruleset, 0)
-
-	// Create a handler closure capable of parsing data from the rulesets endpoint
-	// and appending resultant rulesets to the return slice.
-	responseHandler := func(response *http.Response) (APIListObject, error) {
-		var result ListRulesetsResponse
-		if err := c.decodeJSON(response, &result); err != nil {
-			return APIListObject{}, err
-		}
-
-		rulesets = append(rulesets, result.Rulesets...)
-
-		// Return stats on the current page. Caller can use this information to
-		// adjust for requesting additional pages.
-		return APIListObject{
-			More:   result.More,
-			Offset: result.Offset,
-			Limit:  result.Limit,
-		}, nil
+// ListRulesetsWithOpts get all rulesets with options
+func (c *Client) ListRulesetsWithOpts(o APIListObject) (*ListRulesetsResponse, error) {
```

Comment:
I also forgot, we probably want to update this method to accept a `context.Context` as its first argument, since both `.get()` and `.pagedGet()` now require `context.Context` as their first argument.

Reply:
@kilianw we've aligned on moving forward with adding `context.Context` support to new methods going forward. So before merging we'd need to update this method to take `ctx context.Context` as its first argument, to then pass them to `pagedGet()`. Thanks!